src/main/java/com/iridium/iridiumskyblock/gui/IslandSettingsGUI.java

```diff
@@ -59,15 +58,42 @@ public void onInventoryClick(InventoryClickEvent event) {
             event.getWhoClicked().sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotChangeSettings.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
             return;
         }
-        
-        for (Map.Entry<String, Setting> setting : IridiumSkyblock.getInstance().getSettingsList().entrySet()) {
+
+        for (Map.Entry<String, IslandSettingConfig> setting : IridiumSkyblock.getInstance().getSettingsList().entrySet()) {
             if (event.getSlot() != setting.getValue().getItem().slot) continue;
-            
-            SettingType settingType = SettingType.getByName(setting.getKey());
-            IslandSetting islandSetting = IridiumSkyblock.getInstance().getIslandManager().getIslandSetting(getIsland(), settingType);
-            String newValue = (event.getClick() == ClickType.RIGHT ? settingType.getNext() : settingType.getPrevious()).getNew(islandSetting.getValue());
-            islandSetting.setValue(newValue);
-            settingType.getOnChange().run(getIsland(), newValue);
+
+            IslandSettingType islandSettingType = IslandSettingType.getByName(setting.getKey());
+            IslandSetting islandSetting = IridiumSkyblock.getInstance().getIslandManager().getIslandSetting(getIsland(), islandSettingType);
+            switch (islandSettingType) {
```

Comment:
Issue here is we eventually want to be able to allow people to register custom settings through the GUI, so id like a way to do that here, @dlsf any big brain thoughts on how to do this?

Reply:
I'll probably take a look at it tomorrow

Reply:
Let me this, i'll do :panda:

Reply:
I guess it would be possible to add `IslandSettingType.CUSTOM` for all settings added by external plugins.

Not using the IslandSettingType at all would also be possible, I guess. Just add an `IslandSetting#updateState(InventoryClickEvent)` method or something similar and don't hardcode this.

Thinking about it, the latter option is probably the best one.